src/data/finsupp/witnesses.lean

```diff
@@ -0,0 +1,116 @@
+/-
+Copyright (c) 2022 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.finsupp.basic
+
+/-!
+#  `diff` for finitely supported functions
```

Comment:
There is no `diff` on regular functions, right?

Reply:
Not that I am aware of.  In the case of regular functions, though, you cannot and do not need to produce a finset, so using `{x | f x \ne g x}` is short enough.